From a7e0689c26d2ef7088220ef7d660748271c9274e Mon Sep 17 00:00:00 2001
From: Ryan Schmidt <git@ryandesign.com>
Date: Mon, 8 Oct 2018 19:29:44 -0500
Subject: [PATCH] Always link with libexpat

Always link with libexpat, not just on Linux, GNU Hurd and GNU kFreeBSD.
Fixes build failure on macOS and presumably other non-GNU UNIX systems.

Closes https://github.com/thezbyg/gpick/issues/112

Still only link with librt on Linux, GNU Hurd and GNU kFreeBSD because that
library is specific to the GNU C library.
---
 source/SConscript | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/source/SConscript b/source/SConscript
index a666edc..82ecf83 100644
--- a/source/SConscript
+++ b/source/SConscript
@@ -47,12 +47,10 @@ if local_env['BUILD_TARGET'] == 'win32':
 	else:
 		local_env.Append(LINKFLAGS = ['/SUBSYSTEM:WINDOWS', '/ENTRY:mainCRTStartup'], CPPDEFINES = ['XML_STATIC'])
 	objects.append(SConscript(['winres/SConscript'], exports='env'))
-elif local_env['BUILD_TARGET'] == 'linux2':
-	local_env.Append(LIBS=['rt', 'expat'])
-elif local_env['BUILD_TARGET'].startswith('gnu0'):
-	local_env.Append(LIBS=['rt', 'expat'])
-elif local_env['BUILD_TARGET'].startswith('gnukfreebsd'):
-	local_env.Append(LIBS=['rt', 'expat'])
+else:
+	local_env.Append(LIBS=['expat'])
+	if local_env['BUILD_TARGET'].startswith('linux') or local_env['BUILD_TARGET'].startswith('gnu0') or local_env['BUILD_TARGET'].startswith('gnukfreebsd'):
+		local_env.Append(LIBS=['rt'])
 
 local_env.Append(CPPPATH=['#source'])
 
